src/router.ts

```diff
@@ -47,16 +52,45 @@ router.add('GET', '/rtv/metadata', async (request) => {
   );
 });
 
+/**
+ * Handles amp-geo requests.
+ * @param req - the request object.
+ * @param rtv - RTV number to use.
+ * @param path - path to the requested amp-geo file, must start with `/`.
+ * @returns a dynamically injected amp-geo response, possibly from cache.
+ */
+async function ampGeoRequest(
+  {cf, extend}: ServerRequest,
+  rtv: string,
+  path: string
+): Promise<Response> {
+  const url = getAmpFileUrl(rtv, path);
+  const cacheKey = `${cf.country ?? ''};${cf.regionCode ?? ''}`;
+
+  const cachedResponse = await getCacheFor(url, cacheKey, cf);
+  if (cachedResponse) {
+    console.log('Serving', url, 'with dynamic key', cacheKey, 'from cache');
+    return withHeaders(cachedResponse, CacheControl.AMP_GEO);
```

Comment:
I thought about it, but I think I prefer to inject the correct headers just before the response, whether we fetch this file from the network + inject, from the network cache + inject, or from our own cache. My thoughts were that if we ever do want to make a change to the headers we return, it would be easier to do it this way, which assumes the `cachedResponse`/`response` is incomplete